<a href="https://colab.research.google.com/github/federico2018lc/InnovacionDatosProyectoFinal/blob/main/analisis_datos_csv_(1)_(1).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Etapa preparacion y  extraccion: Para comenzar el analisis, importamos las librerias a utilizar y cargamos los archivos csv para luego realizar etl

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
clientes=pd.read_csv( 'customer_data.csv')
ventas= pd.read_csv('sales_data.csv')


In [ ]:
clientes


customer_id  gender   age payment_method
0         C241288  Female  28.0    Credit Card
1         C111565    Male  21.0     Debit Card
2         C266599    Male  20.0           Cash
3         C988172  Female  66.0    Credit Card
4         C189076  Female  53.0           Cash
...           ...     ...   ...            ...
99452     C441542  Female  45.0    Credit Card
99453     C569580    Male  27.0           Cash
99454     C103292    Male  63.0     Debit Card
99455     C800631    Male  56.0           Cash
99456     C273973  Female  36.0    Credit Card

[99457 rows x 4 columns]

In [ ]:
ventas


invoice_no customer_id         category  quantity    price invoice_date  \
0        I138884     C241288         Clothing         5  1500.40   05-08-2022   
1        I317333     C111565            Shoes         3  1800.51   12-12-2021   
2        I127801     C266599         Clothing         1   300.08   09-11-2021   
3        I173702     C988172            Shoes         5  3000.85   16-05-2021   
4        I337046     C189076            Books         4    60.60   24-10-2021   
...          ...         ...              ...       ...      ...          ...   
99452    I219422     C441542         Souvenir         5    58.65   21-09-2022   
99453    I325143     C569580  Food & Beverage         2    10.46   22-09-2021   
99454    I824010     C103292  Food & Beverage         2    10.46   28-03-2021   
99455    I702964     C800631       Technology         4  4200.00   16-03-2021   
99456    I232867     C273973         Souvenir         3    35.19   15-10-2022   

          shopping_mall  
0                Kanyon  
1        Forum Istanbul  
2             Metrocity  
3          Metropol AVM  
4                Kanyon  
...                 ...  
99452            Kanyon  
99453    Forum Istanbul  
99454         Metrocity  
99455      Istinye Park  
99456  Mall of Istanbul  

[99457 rows x 7 columns]

In [ ]:
clientes.head()


customer_id  gender   age payment_method
0     C241288  Female  28.0    Credit Card
1     C111565    Male  21.0     Debit Card
2     C266599    Male  20.0           Cash
3     C988172  Female  66.0    Credit Card
4     C189076  Female  53.0           Cash

In [ ]:
ventas.head()

invoice_no customer_id  category  quantity    price invoice_date  \
0    I138884     C241288  Clothing         5  1500.40   05-08-2022   
1    I317333     C111565     Shoes         3  1800.51   12-12-2021   
2    I127801     C266599  Clothing         1   300.08   09-11-2021   
3    I173702     C988172     Shoes         5  3000.85   16-05-2021   
4    I337046     C189076     Books         4    60.60   24-10-2021   

    shopping_mall  
0          Kanyon  
1  Forum Istanbul  
2       Metrocity  
3    Metropol AVM  
4          Kanyon

In [ ]:
clientes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99457 entries, 0 to 99456
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   customer_id     99457 non-null  object 
 1   gender          99457 non-null  object 
 2   age             99338 non-null  float64
 3   payment_method  99457 non-null  object 
dtypes: float64(1), object(3)
memory usage: 3.0+ MB


In [ ]:
clientes.info()

<class 'pandas.core.frame.DataFrame'>
Index: 99338 entries, C241288 to C273973
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   gender          99338 non-null  object
 1   age             99338 non-null  int64 
 2   payment_method  99338 non-null  object
dtypes: int64(1), object(2)
memory usage: 5.0+ MB


In [ ]:
ventas.info()

<class 'pandas.core.frame.DataFrame'>
Index: 99457 entries, C241288 to C273973
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   invoice_no     99457 non-null  object        
 1   category       99457 non-null  object        
 2   quantity       99457 non-null  int64         
 3   price          99457 non-null  float64       
 4   invoice_date   99457 non-null  datetime64[ns]
 5   shopping_mall  99457 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(3)
memory usage: 5.3+ MB


con las sentencias anteriores, pudimos observar que tipo de datos contiene cada data set ( objects: innvoice no, customerid, gender, payment method, category, invoice date, shopping mall), float (age, price) e int (quantity). Observamos que tienen en comun la columna customer_id y que las primeras columnas pueden ser eliminadas(contienen un indice) ya que usaremos la columna id

Etapa: transformación y limpieza de datos

In [ ]:
clientes=pd.read_csv( 'customer_data.csv',index_col='customer_id') # ordendo por columna customer id

In [ ]:
clientes

gender   age payment_method
customer_id                             
C241288      Female  28.0    Credit Card
C111565        Male  21.0     Debit Card
C266599        Male  20.0           Cash
C988172      Female  66.0    Credit Card
C189076      Female  53.0           Cash
...             ...   ...            ...
C441542      Female  45.0    Credit Card
C569580        Male  27.0           Cash
C103292        Male  63.0     Debit Card
C800631        Male  56.0           Cash
C273973      Female  36.0    Credit Card

[99457 rows x 3 columns]

In [ ]:
ventas= pd.read_csv('sales_data.csv', index_col= 'customer_id') # ordendo por columna customer id

In [ ]:
ventas

invoice_no         category  quantity    price invoice_date  \
customer_id                                                               
C241288        I138884         Clothing         5  1500.40   05-08-2022   
C111565        I317333            Shoes         3  1800.51   12-12-2021   
C266599        I127801         Clothing         1   300.08   09-11-2021   
C988172        I173702            Shoes         5  3000.85   16-05-2021   
C189076        I337046            Books         4    60.60   24-10-2021   
...                ...              ...       ...      ...          ...   
C441542        I219422         Souvenir         5    58.65   21-09-2022   
C569580        I325143  Food & Beverage         2    10.46   22-09-2021   
C103292        I824010  Food & Beverage         2    10.46   28-03-2021   
C800631        I702964       Technology         4  4200.00   16-03-2021   
C273973        I232867         Souvenir         3    35.19   15-10-2022   

                shopping_mall  
customer_id                    
C241288                Kanyon  
C111565        Forum Istanbul  
C266599             Metrocity  
C988172          Metropol AVM  
C189076                Kanyon  
...                       ...  
C441542                Kanyon  
C569580        Forum Istanbul  
C103292             Metrocity  
C800631          Istinye Park  
C273973      Mall of Istanbul  

[99457 rows x 6 columns]

In [ ]:
clientes.isnull().sum()

customer_id       0
gender            0
age               0
payment_method    0
dtype: int64

In [ ]:
clientes.dropna (subset=['age'], inplace=True)#elimino los 119 registros vacios, ya que esa cantidad no afecta la muestra de 99338 valores
clientes['age'] = clientes['age'].astype(int)  # convierto a enteros la edad

In [ ]:
clientes.info()

<class 'pandas.core.frame.DataFrame'>
Index: 99338 entries, 0 to 99456
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   customer_id     99338 non-null  object
 1   gender          99338 non-null  object
 2   age             99338 non-null  int64 
 3   payment_method  99338 non-null  object
dtypes: int64(1), object(3)
memory usage: 3.8+ MB


In [ ]:
ventas = ventas[ventas['customer_id'].isin(clientes['customer_id'])]  # elimino esos 119 regristros de ventas

In [ ]:
ventas

invoice_no customer_id         category  quantity    price invoice_date  \
0        I138884     C241288         Clothing         5  1500.40   2022-08-05   
1        I317333     C111565            Shoes         3  1800.51   2021-12-12   
2        I127801     C266599         Clothing         1   300.08   2021-11-09   
3        I173702     C988172            Shoes         5  3000.85   2021-05-16   
4        I337046     C189076            Books         4    60.60   2021-10-24   
...          ...         ...              ...       ...      ...          ...   
99452    I219422     C441542         Souvenir         5    58.65   2022-09-21   
99453    I325143     C569580  Food & Beverage         2    10.46   2021-09-22   
99454    I824010     C103292  Food & Beverage         2    10.46   2021-03-28   
99455    I702964     C800631       Technology         4  4200.00   2021-03-16   
99456    I232867     C273973         Souvenir         3    35.19   2022-10-15   

          shopping_mall  
0                Kanyon  
1        Forum Istanbul  
2             Metrocity  
3          Metropol AVM  
4                Kanyon  
...                 ...  
99452            Kanyon  
99453    Forum Istanbul  
99454         Metrocity  
99455      Istinye Park  
99456  Mall of Istanbul  

[99338 rows x 7 columns]

In [ ]:
ventas.isnull().sum()

invoice_no       0
customer_id      0
category         0
quantity         0
price            0
invoice_date     0
shopping_mall    0
dtype: int64

In [ ]:
ventas.loc[:,'invoice_date']= pd.to_datetime(ventas['invoice_date'], format='%d-%m-%Y') #estandarizo fechas. Uso loc para asegurar que se modifique toda la columna del df original

In [ ]:
ventas.head() #confirmo si el cambio se realizó ok


invoice_no customer_id  category  quantity    price invoice_date  \
0    I138884     C241288  Clothing         5  1500.40   2022-08-05   
1    I317333     C111565     Shoes         3  1800.51   2021-12-12   
2    I127801     C266599  Clothing         1   300.08   2021-11-09   
3    I173702     C988172     Shoes         5  3000.85   2021-05-16   
4    I337046     C189076     Books         4    60.60   2021-10-24   

    shopping_mall  
0          Kanyon  
1  Forum Istanbul  
2       Metrocity  
3    Metropol AVM  
4          Kanyon

In [ ]:
ventas.info() # confirmo si el tipo de dato dejo de ser un object

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99457 entries, 0 to 99456
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   invoice_no     99457 non-null  object        
 1   customer_id    99457 non-null  object        
 2   category       99457 non-null  object        
 3   quantity       99457 non-null  int64         
 4   price          99457 non-null  float64       
 5   invoice_date   99457 non-null  datetime64[ns]
 6   shopping_mall  99457 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(4)
memory usage: 5.3+ MB


In [ ]:
(clientes['gender'].value_counts(normalize=True)*100).round(2) # aqui vemos cantidad de mujeres y hombres de la muestra en porcentajes


gender
Female    59.81
Male      40.19
Name: proportion, dtype: float64

In [ ]:
clientes ['age'].describe() # vemos promedio de edades (43.4 años), desviacion estandar (14.98), max (69) y min ( 18 años)

count    99338.000000
mean        43.425859
std         14.989400
min         18.000000
25%         30.000000
50%         43.000000
75%         56.000000
max         69.000000
Name: age, dtype: float64

In [ ]:
(clientes ['payment_method'].value_counts(normalize=True)*100).round(2) #aqui vemos metodos de pagos diferenciados en porcentajes

payment_method
Cash           44.69
Credit Card    35.13
Debit Card     20.18
Name: proportion, dtype: float64

In [ ]:
(clientes.groupby('gender')['payment_method'].value_counts(normalize=True)*100).round(2)

gender  payment_method
Female  Cash              44.57
        Credit Card       35.34
        Debit Card        20.10
Male    Cash              44.88
        Credit Card       34.82
        Debit Card        20.30
Name: proportion, dtype: float64

In [ ]:
ventas.describe()


quantity         price                invoice_date
count  99457.000000  99457.000000                       99457
mean       3.003429    689.256321  2022-02-04 02:46:59.783424
min        1.000000      5.230000         2021-01-01 00:00:00
25%        2.000000     45.450000         2021-07-19 00:00:00
50%        3.000000    203.300000         2022-02-05 00:00:00
75%        4.000000   1200.320000         2022-08-22 00:00:00
max        5.000000   5250.000000         2023-03-08 00:00:00
std        1.413025    941.184567                         NaN

In [ ]:
(ventas ['category'].value_counts(normalize=True)*100).round(2)


category
Clothing           34.68
Cosmetics          15.18
Food & Beverage    14.86
Toys               10.14
Shoes              10.09
Souvenir            5.03
Technology          5.02
Books               5.01
Name: proportion, dtype: float64

In [ ]:
ventas ['price'].describe() # aqui vemos info sobre los precios, prmedio, desviacion standar, max y min

count    99457.000000
mean       689.256321
std        941.184567
min          5.230000
25%         45.450000
50%        203.300000
75%       1200.320000
max       5250.000000
Name: price, dtype: float64

In [ ]:
(ventas ['shopping_mall'].value_counts(normalize=True)*100).round(2) # aqui vemos %de participacion de cada mall

shopping_mall
Mall of Istanbul     20.05
Kanyon               19.93
Metrocity            15.09
Metropol AVM         10.22
Istinye Park          9.83
Zorlu Center          5.10
Cevahir AVM           5.02
Forum Istanbul        4.97
Viaport Outlet        4.94
Emaar Square Mall     4.84
Name: proportion, dtype: float64

In [ ]:
ventas['invoice_date'].min(), ventas['invoice_date'].max() # aca vemos rango de fechas de la muestra

(Timestamp('2021-01-01 00:00:00'), Timestamp('2023-03-08 00:00:00'))

En este paso, unimos los dos data frame, utilizando columna customer_id

In [ ]:
df_final=pd.merge(ventas, clientes, on='customer_id', how='inner')

In [ ]:
df_final

invoice_no         category  quantity    price invoice_date  \
customer_id                                                               
C241288        I138884         Clothing         5  1500.40   2022-08-05   
C111565        I317333            Shoes         3  1800.51   2021-12-12   
C266599        I127801         Clothing         1   300.08   2021-11-09   
C988172        I173702            Shoes         5  3000.85   2021-05-16   
C189076        I337046            Books         4    60.60   2021-10-24   
...                ...              ...       ...      ...          ...   
C441542        I219422         Souvenir         5    58.65   2022-09-21   
C569580        I325143  Food & Beverage         2    10.46   2021-09-22   
C103292        I824010  Food & Beverage         2    10.46   2021-03-28   
C800631        I702964       Technology         4  4200.00   2021-03-16   
C273973        I232867         Souvenir         3    35.19   2022-10-15   

                shopping_mall  gender   age payment_method  
customer_id                                                 
C241288                Kanyon  Female  28.0    Credit Card  
C111565        Forum Istanbul    Male  21.0     Debit Card  
C266599             Metrocity    Male  20.0           Cash  
C988172          Metropol AVM  Female  66.0    Credit Card  
C189076                Kanyon  Female  53.0           Cash  
...                       ...     ...   ...            ...  
C441542                Kanyon  Female  45.0    Credit Card  
C569580        Forum Istanbul    Male  27.0           Cash  
C103292             Metrocity    Male  63.0     Debit Card  
C800631          Istinye Park    Male  56.0           Cash  
C273973      Mall of Istanbul  Female  36.0    Credit Card  

[99338 rows x 9 columns]

In [ ]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
Index: 99338 entries, C241288 to C273973
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   invoice_no      99338 non-null  object        
 1   category        99338 non-null  object        
 2   quantity        99338 non-null  int64         
 3   price           99338 non-null  float64       
 4   invoice_date    99338 non-null  datetime64[ns]
 5   shopping_mall   99338 non-null  object        
 6   gender          99338 non-null  object        
 7   age             99338 non-null  int64         
 8   payment_method  99338 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(5)
memory usage: 9.6+ MB


In [ ]:
 df_final['age'] = df_final['age'].astype(int)

In [ ]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
Index: 99338 entries, C241288 to C273973
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   invoice_no      99338 non-null  object        
 1   category        99338 non-null  object        
 2   quantity        99338 non-null  int64         
 3   price           99338 non-null  float64       
 4   invoice_date    99338 non-null  datetime64[ns]
 5   shopping_mall   99338 non-null  object        
 6   gender          99338 non-null  object        
 7   age             99338 non-null  int64         
 8   payment_method  99338 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(5)
memory usage: 9.6+ MB


Ahora si, estamos en condiciones de responder preguntas de la consigna


1 Metodo de pago mas usado:  en est caso, se observa que es el Cash ( efectivo)

In [ ]:
df_final['payment_method'].mode()

0    Cash
Name: payment_method, dtype: object

2 Método de pago por genero: aqui se observa que tanto hombres como mujeres utilizan en similares porcentajes los medios de pago, siendo el mas usado el cash( 44,57% mujeres - 44.88% hombres), luego credito (35.34% mujeres - 34.82% hombres) y finalmente debito ( 20.1% mujeres - 20.3% hombres)

In [ ]:
df_final.groupby('gender')['payment_method'].value_counts()


gender  payment_method
Female  Cash              26479
        Credit Card       20994
        Debit Card        11939
Male    Cash              17918
        Credit Card       13904
        Debit Card         8104
Name: count, dtype: int64

In [ ]:
(df_final.groupby('gender')['payment_method'].value_counts(normalize=True)*100).round(2)

gender  payment_method
Female  Cash              44.57
        Credit Card       35.34
        Debit Card        20.10
Male    Cash              44.88
        Credit Card       34.82
        Debit Card        20.30
Name: proportion, dtype: float64

3 categorización de clientes de acuerdo a su forma de pago (por
edad y género).

In [ ]:
categorizacion= (
    df_final
    .groupby(['age', 'gender', 'payment_method'])  # agrupo los datos edad, genero y metodo de pago
    .size()                                        #cuenta cuantos registros hay en cada grupp
    .reset_index(name='Cantidad')                  #conviete el resultado en un df y lo llma cantidad
    .sort_values(by=['age', 'gender'])              # ordena
)

In [ ]:
categorizacion

age  gender payment_method  Cantidad
0     18  Female           Cash       473
1     18  Female    Credit Card       383
2     18  Female     Debit Card       222
3     18    Male           Cash       342
4     18    Male    Credit Card       267
..   ...     ...            ...       ...
307   69  Female    Credit Card       394
308   69  Female     Debit Card       217
309   69    Male           Cash       325
310   69    Male    Credit Card       289
311   69    Male     Debit Card       143

[312 rows x 4 columns]

4 Métodos de pagos realizados por el rango etario de 25 a 35 años

In [ ]:
df_25_35= df_final[ (df_final['age']>=25)& (df_final['age'] <=35)]
(df_25_35 ['payment_method'].value_counts(normalize=True)*100).round(2)

payment_method
Cash           44.78
Credit Card    34.82
Debit Card     20.40
Name: proportion, dtype: float64

##### 5- Métodos de pago más utilizados por las mujeres. Se observa que el 44.57% utiliza cash

In [ ]:
(df_final[df_final['gender']=='Female']['payment_method'].value_counts(normalize=True)*100).round(2)

payment_method
Cash           44.57
Credit Card    35.34
Debit Card     20.10
Name: proportion, dtype: float64

6- Precios por categoría de productos: aqui vemos el listado de categorias de prodcutos y sus precios promedio

In [ ]:
(df_final.groupby('category')['price'].mean()).round(2)

category
Books                45.56
Clothing            901.12
Cosmetics           122.45
Food & Beverage      15.67
Shoes              1807.28
Souvenir             34.88
Technology         3157.15
Toys                107.75
Name: price, dtype: float64

ETAPA LOAD - datos limpios. Guardamos datos finales para hacer el analisis

In [ ]:
datos_limpios=df_final.copy()   # creo un data frame limpio

In [ ]:
datos_limpios

invoice_no         category  quantity    price invoice_date  \
customer_id                                                               
C241288        I138884         Clothing         5  1500.40   2022-08-05   
C111565        I317333            Shoes         3  1800.51   2021-12-12   
C266599        I127801         Clothing         1   300.08   2021-11-09   
C988172        I173702            Shoes         5  3000.85   2021-05-16   
C189076        I337046            Books         4    60.60   2021-10-24   
...                ...              ...       ...      ...          ...   
C441542        I219422         Souvenir         5    58.65   2022-09-21   
C569580        I325143  Food & Beverage         2    10.46   2021-09-22   
C103292        I824010  Food & Beverage         2    10.46   2021-03-28   
C800631        I702964       Technology         4  4200.00   2021-03-16   
C273973        I232867         Souvenir         3    35.19   2022-10-15   

                shopping_mall  gender  age payment_method  
customer_id                                                
C241288                Kanyon  Female   28    Credit Card  
C111565        Forum Istanbul    Male   21     Debit Card  
C266599             Metrocity    Male   20           Cash  
C988172          Metropol AVM  Female   66    Credit Card  
C189076                Kanyon  Female   53           Cash  
...                       ...     ...  ...            ...  
C441542                Kanyon  Female   45    Credit Card  
C569580        Forum Istanbul    Male   27           Cash  
C103292             Metrocity    Male   63     Debit Card  
C800631          Istinye Park    Male   56           Cash  
C273973      Mall of Istanbul  Female   36    Credit Card  

[99338 rows x 9 columns]

Antes de guardarlo como definitivo, hacemos comprobaciones de integridad ( no nulos, no duplicados,tipos de datos)

In [ ]:
datos_limpios[['age', 'payment_method','gender', 'price', 'shopping_mall']].isnull().sum()

age               0
payment_method    0
gender            0
price             0
shopping_mall     0
dtype: int64

In [ ]:
datos_limpios.index.isnull().sum() #acá nos fijamos que customer_id no sea nulo

np.int64(0)

In [ ]:
datos_limpios.duplicated().sum()

np.int64(0)

In [ ]:
set(ventas['customer_id']).issubset(set(clientes['customer_id']))# aca chequeamos integridad referencial ( mismos registros  en ventas que en clientes)

True

In [ ]:
datos_limpios.info()  #para ver tipos de datos

<class 'pandas.core.frame.DataFrame'>
Index: 99338 entries, C241288 to C273973
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   invoice_no      99338 non-null  object        
 1   category        99338 non-null  object        
 2   quantity        99338 non-null  int64         
 3   price           99338 non-null  float64       
 4   invoice_date    99338 non-null  datetime64[ns]
 5   shopping_mall   99338 non-null  object        
 6   gender          99338 non-null  object        
 7   age             99338 non-null  int64         
 8   payment_method  99338 non-null  object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(5)
memory usage: 9.6+ MB


Una vez verificado la integridad de los datos, se genera archivo datos_limplios.csv

In [ ]:
datos_limpios.to_csv('datos_limpios.csv', index=False)

verificaciones finales:

In [ ]:
df_final=pd.read_csv('datos_limpios.csv')

In [ ]:
df_final

invoice_no         category  quantity    price invoice_date  \
0        I138884         Clothing         5  1500.40   2022-08-05   
1        I317333            Shoes         3  1800.51   2021-12-12   
2        I127801         Clothing         1   300.08   2021-11-09   
3        I173702            Shoes         5  3000.85   2021-05-16   
4        I337046            Books         4    60.60   2021-10-24   
...          ...              ...       ...      ...          ...   
99333    I219422         Souvenir         5    58.65   2022-09-21   
99334    I325143  Food & Beverage         2    10.46   2021-09-22   
99335    I824010  Food & Beverage         2    10.46   2021-03-28   
99336    I702964       Technology         4  4200.00   2021-03-16   
99337    I232867         Souvenir         3    35.19   2022-10-15   

          shopping_mall  gender  age payment_method  
0                Kanyon  Female   28    Credit Card  
1        Forum Istanbul    Male   21     Debit Card  
2             Metrocity    Male   20           Cash  
3          Metropol AVM  Female   66    Credit Card  
4                Kanyon  Female   53           Cash  
...                 ...     ...  ...            ...  
99333            Kanyon  Female   45    Credit Card  
99334    Forum Istanbul    Male   27           Cash  
99335         Metrocity    Male   63     Debit Card  
99336      Istinye Park    Male   56           Cash  
99337  Mall of Istanbul  Female   36    Credit Card  

[99338 rows x 9 columns]

In [ ]:
df_final.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99338 entries, 0 to 99337
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   invoice_no      99338 non-null  object 
 1   category        99338 non-null  object 
 2   quantity        99338 non-null  int64  
 3   price           99338 non-null  float64
 4   invoice_date    99338 non-null  object 
 5   shopping_mall   99338 non-null  object 
 6   gender          99338 non-null  object 
 7   age             99338 non-null  int64  
 8   payment_method  99338 non-null  object 
dtypes: float64(1), int64(2), object(6)
memory usage: 6.8+ MB


Aca viene la parte de analisis: ver distribucion de edades,ventas segun metodos de pago, ventas segun genero y metodo de pago.  Esto sirve para el informe final y conclusiones
